.github/workflows/rust.yml

```diff
@@ -22,144 +22,300 @@ on:
+    needs: [linux-build-lib]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64]
+        rust: [nightly-2020-11-24]
+    container:
+      image: docker.pkg.github.com/${{ github.repository }}/${{ matrix.arch }}-linux-rust-${{ matrix.rust }}
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+      env:
+        CARGO_HOME: /github/home/.cargo
+        CARGO_TARGET_DIR: /github/home/target
+
+        ARROW_TEST_DATA: /__w/arrow/arrow/testing/data
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: /github/home/.cargo
+          # these keys equal the ones on `linux-build-lib` for re-use
+          key: cargo-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: cargo-cache-
+      - name: Cache Rust dependencies
+        uses: actions/cache@v2
+        with:
+          path: /github/home/target
+          # these keys equal the ones on `linux-build-lib` for re-use
+          key: ${{ runner.os }}-${{ matrix.arch }}-target-cache-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-${{ matrix.arch }}-target-cache-${{ matrix.rust }}-
+      - name: Run tests
+        run: |
+          cd rust/arrow
+          cargo test --features "simd"
+
+  # test the arrow crate with stable rust
+  linux-test-stable:
+    name: Linux ${{ matrix.arch }} Rust ${{ matrix.rust }} test arrow
+    needs: [linux-build-image]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64]
+        rust: [stable]
+    container:
+      image: docker.pkg.github.com/${{ github.repository }}/${{ matrix.arch }}-linux-rust-${{ matrix.rust }}
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+      env:
+        CARGO_HOME: /github/home/.cargo
+        CARGO_TARGET_DIR: /github/home/target
+
+        ARROW_TEST_DATA: /__w/arrow/arrow/testing/data
     steps:
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: /github/home/.cargo
+          # these keys equal the ones on `linux-build-lib` for re-use
+          key: cargo-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: cargo-cache-
+      - name: Cache Rust dependencies
+        uses: actions/cache@v2
+        with:
+          path: /github/home/target
+          # these keys equal the ones on `linux-build-lib` for re-use
+          key: ${{ runner.os }}-${{ matrix.arch }}-target-cache-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-${{ matrix.arch }}-target-cache-${{ matrix.rust }}-
+      - name: Run tests
+        run: |
+          cd rust/arrow
```

Comment:
@GregBowyer we'd be able to add parquet to stable tests here, so we can leave out adding it to CI in your PR